File: online_judge/lintcode/climbing_stairs.py
class Solution(object):

    def climbStairs(self, n):
        if n < 2:
            return 1
        f = [0] * (n+1)
        f[0] = 1
        f[1] = 1
        for i in range(2, n+1):
            f[i] = f[i-1] + f[i-2]

        return f[n]

File: online_judge/leetcode/climbing-stairs.cc
#include <iostream>
#include <vector>

using namespace std;

class Solution {
public:
    int climbStairs(int n) {
        int p[2] = {1, 2};
        if (n < 3) return p[n-1];

        int pi = 0;
        for (int i = 2; i < n; ++i) {
            pi = p[0] + p[1];
            p[0] = p[1];
            p[1] = pi;
        }
        return pi;
    }
};
